src/snapred/backend/recipe/algorithm/SmoothDataExcludingPeaksAlgo.py

```diff
@@ -65,19 +65,22 @@ def PyExec(self):
             WeightWorkspace=weight_ws_name,
         )
         self.mantidSnapper.executeQueue()
-        weights_ws = mtd[weight_ws_name]
 
         # create workspace group
-        ws_group = WorkspaceGroup()
-        mtd.add("SmoothedDataExcludingPeaks", ws_group)
+        if mtd.doesExist("SmoothedDataExcludingPeaks"):
+            ws_group = mtd["SmoothedDataExcludingPeaks"]
+        else:
+            ws_group = WorkspaceGroup()
+            mtd.add("SmoothedDataExcludingPeaks", ws_group)
 
         # get number of spectrum to iterate over
-        numSpec = weights_ws.getNumberHistograms()
+
+        numSpec = mtd[weight_ws_name].getNumberHistograms()
 
         # extract x & y data for csaps
         for index in range(numSpec):
-            x = weights_ws.readX(index)  # len of 1794
-            w = weights_ws.readY(index)  # len of 1793
+            x = mtd[weight_ws_name].readX(index)  # len of 1794
+            w = mtd[weight_ws_name].readY(index)  # len of 1793
```

Comment:
Wouldn't it be better to grab the handle to `weight_ws_name` to avoid this repeated access to the ADS?

Reply:
I'm not sure why but if I do 
```
weight_ws = mtd[weight_ws_name]
x = weight_ws.readX(index)  
w = weight_ws.readY(index) 
```
It throws an error that the workspace has been invalidated